scripts/updateRecordsCatalogLink.py

```diff
@@ -0,0 +1,43 @@
+import os
+import re
+
+from model import Record
+from main import loadEnvFile
+from managers import DBManager
+
+
+def main():
+    '''
+    Updating NYPL catalog records with new link to replace text/html link
+    '''
+
+    loadEnvFile('local-qa', fileString='config/{}.yaml')
+
+    dbManager = DBManager(
+        user= os.environ.get('POSTGRES_USER', None),
+        pswd= os.environ.get('POSTGRES_PSWD', None),
+        host= os.environ.get('POSTGRES_HOST', None),
+        port= os.environ.get('POSTGRES_PORT', None),
+        db= os.environ.get('POSTGRES_NAME', None)
+    )
+
+    dbManager.generateEngine()
+
+    dbManager.createSession()
+
+    catalogRegex = r'catalog\": true' 
+    linkRegex = r'text\/html'
+
+    for record in dbManager.session.query(Record) \
+        .filter(Record.source == 'nypl').all():
+            
+            for i, elem in enumerate(record.has_part):
+                if re.search(linkRegex, elem) != None:
+                    if re.search(catalogRegex, elem) != None:
+                        record.has_part = [record.has_part[i].replace('text/html', 'application/html+catalog')]
```

Comment:
This will have a side effect of resetting the `has_part` array to a single value, deleting any other values originally in the array!

I think the best way to deal with this is to create a local array that all of the `has_part` strings get loaded into, and then set the column value to that after the loop is executed.

Reply:
I believe putting record.has_part[i].replace('text/html', 'application/html+catalog') into an array fixed this issue since when I tested out my script on the Record with the id 4924, the has_part field updated with the new catalog media type.

`{"1|https://www.nypl.org/research/collections/shared-collection-catalog/bib/b22313364|nypl|application/html+catalog|{\"catalog\": true, \"download\": false, \"reader\": false}"}`

The has_part field for that record was originally this:

`{"1|https://www.nypl.org/research/collections/shared-collection-catalog/bib/b22313364|nypl|text/html|{\"catalog\": true, \"download\": false, \"reader\": false}"}`